### PR TITLE
Amend NFT ID resolves #303 

### LINKED ISSUE #303: Add ID to the NFT UTXO model
### Problem to solve

In Bitcoin UTXO model, to identify which UTXO from which transaction we are using, a transaction ID and UTXO ID is used.

Also, in order to support multiple NFT UTXO and specific consumption of them, we need a way to identify the one to use.

This is particularly necessary in NFT collection minting, where we are using UTXO for each token and want to send them individually to other wallets.

### Solution

The proposal would be to add an UTXO's ID and NFT's ID in the transaction data structure to resolve this point.
So a transaction could be built to use a specific NFT UTXO to consume.

This would involve:
- [x] Update transaction data structure for NFT Ledger and UTXO
- [x] Update the ledger operations consumption algorithm

### PR DESCRIPTION
**Pull request by apoorv-2204**
# Description

Fixes # (#303)

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# How Has This Been Tested?

- [x] doc tests
- [x] mix archethic. Regression 

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/archethic/contracts/interpreter.ex [hunk 1]
```diff
@@ -697,6 +697,7 @@ defmodule Archethic.Contracts.Interpreter do
            {{:atom, "to"}, _to},
            {{:atom, "amount"}, _amount},
            {{:atom, "nft"}, _nft_address}
+           # ??
```
Comment (apoorv-2204) [inline lib/archethic/contracts/interpreter.ex:700]: I don't get what to do here? Please brief me about prewalk. {}

Comment (samuel-uniris) [reply lib/archethic/contracts/interpreter.ex:700]: You should add a parameter to accept nft_id in the interpreter parser

File: lib/archethic_web/graphql_schema/transaction_type.ex [hunk 2]
```diff
@@ -78,6 +78,7 @@ defmodule ArchethicWeb.GraphQLSchema.TransactionType do
     field(:to, :address)
     field(:amount, :amount)
     field(:nft, :address)
+    field(:nft_id, :nft_id)
```
Comment (apoorv-2204) [inline lib/archethic_web/graphql_schema/transaction_type.ex:81]: I am not so good with graphql, I am getting this error and any suggestions?```= Compilation error in file lib/archethic_web/graphql_schema.ex ==
** (Absinthe.Schema.Error) Compilation failed:
---------------------------------------
## Locations
/home/apoorv/Documents/aenode/lib/archethic_web/graphql_schema/transaction_type.ex:81

In Nft_id, :nft_id is not defined in your schema.

Types must exist if referenced.
---------------------------------------
## Locations
/home/apoorv/Documents/aenode/lib/archethic_web/graphql_schema/transaction_type.ex:162

In Nft_id, :nft_id is not defined in your schema.

Types must exist if referenced.
---------------------------------------
## Locations
/home/apoorv/Documents/aenode/lib/archethic_web/graphql_schema/transaction_type.ex:198

In Nft_id, :nft_id is not defined in your schema.

Types must exist if referenced.
---------------------------------------
## Locations
/home/apoorv/Documents/aenode/lib/archethic_web/graphql_schema/transaction_type.ex:182

In Nft_id, :nft_id is not defined in your schema.

Types must exist if referenced.
---------------------------------------
## Locations
/home/apoorv/Documents/aenode/lib/archethic_web/graphql_schema/transaction_type.ex:232

In Nft_id, :nft_id is not defined in your schema.

Types must exist if referenced.```

Comment (apoorv-2204) [reply lib/archethic_web/graphql_schema/transaction_type.ex:81]: I have gone through code, but couldn't find out about  schema , where to mention nft)id

Comment (samuel-uniris) [reply lib/archethic_web/graphql_schema/transaction_type.ex:81]: The NFT id type doesn't exist.
I don't think you need a custom type for nft id, just use an integer

**Comment by apoorv-2204** [comment]
we might need to change the merge branch to develop.

**Pull request closed without merging**